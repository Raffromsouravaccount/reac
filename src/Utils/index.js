import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';
import rootReducer from '../_reducers'
import thunkMiddleware from 'redux-thunk';

/**
 * Create a testing store with import reducer, middleware, and initial state
 * @param {object} component
 * @param {string} attr
 * @function findByTestAttr
 * @returns {wrapper}
 */
export const findByTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

/**
 * Create a testing store with import reducer, middleware, and initial state
 * @param {object} component
 * @param {Object} expectedProps
 * @function checkProps
 * @returns {propsErr}
 */
export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
  return propsErr;
};


/**
 * Create a testing store with import reducer, middleware, and initial state
 * globals: rootReducer
 * @param {object} initialState
 * @function storeFactory
 * @returns {Store} - Redux store
 */
export const storeFactory = (initialState) => {
  const middleWares = [thunkMiddleware];
  const createStoreWithMiddleware = applyMiddleware(...middleWares)(createStore)
  return createStoreWithMiddleware(rootReducer, initialState)
}
