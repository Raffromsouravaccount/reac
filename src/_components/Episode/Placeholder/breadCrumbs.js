import {
  constantText
} from "../../../_helpers/constants.text";

export const breadCrumbs = {
  links: (tvShowUrl, seasonUrl, tvShowId, seasonId) => [{
    color: 'inherit',
    text: constantText.placeholderConstant.dashboard,
    route: '/dashboard'
  },
  {
    color: 'inherit',
    text: constantText.placeholderConstant.shows,
    route: '/tvshow'
  },
  {
    color: 'inherit',
    text: constantText.placeholderConstant.tvShow,
    route: `/tvshow${tvShowUrl}/${tvShowId}`
  },
  {
    color: 'inherit',
    text: constantText.placeholderConstant.seasons,
    route: `/tvshow${tvShowUrl}/${tvShowId}/season`
  },
  {
    color: 'inherit',
    text: constantText.placeholderConstant.season,
    route: `/tvshow${tvShowUrl}/${tvShowId}/season${seasonUrl}/${seasonId}`
  },
  {
    color: 'inherit',
    text: constantText.placeholderConstant.episode,
    route: `/tvshow${tvShowUrl}/${tvShowId}/season${seasonUrl}/${seasonId}/episode`
  }
],
  typography: [{
    color: 'textPrimary',
    text: constantText.placeholderConstant.placeholder
  }]
};

