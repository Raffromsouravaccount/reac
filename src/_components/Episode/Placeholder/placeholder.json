{
  "placeholderProperties": [
    {
      "name": "assetType",
      "value": null,
      "col": "col-md-6 col-lg-6",
      "type": "dropdownAsync",
      "multiple": false,
      "label": "Asset Type",
      "keyText": "title",
      "path": "/master/EpisodeSubType",
      "data": [],
      "errorText": "",
      "validation": {
        "required": true
      }
    },
    {
      "name": "episodeNoStartFrom",
      "value": "",
      "col": "col-md-6 col-lg-6",
      "type": "text",
      "label": "Asset No start from",
      "helperText": "",
      "errorText": "",
      "validation": {
        "required": true,
        "isDigit": true
      }
    },
    {
      "name": "dayPart",
      "value": null,
      "col": "col-md-6 col-lg-6",
      "type": "dropdown",
      "multiple": true,
      "label": "Day Part",
      "helperText": "",
      "keyText": "title",
      "groupBy":"group",
      "data": [],
      "errorText": "",
      "validation": {
        "required": false
      }
    },
    {
      "name": "numberOfAssets",
      "value": "",
      "col": "col-md-6 col-lg-6",
      "type": "text",
      "label": "Number of Assets",
      "errorText": "",
      "validation": {
        "required": true,
        "maxValue": 50,
        "isDigit": true
      }
    },
    {
      "name": "assetsPerDay",
      "value": "",
      "col": "col-md-6 col-lg-6",
      "type": "text",
      "label": "Assets Per Day",
      "disabled": true,
      "errorText": "",
      "validation": {
        "required": false,
        "maxValue": 150,
        "isDigit": true
      }
    },

    {
      "name": "telecastDate",
      "type": "date",
      "col": "col-md-6 col-lg-6",
      "value": "",
      "label": "Telecast Date",
      "minDateValue": "",
      "helperText": "",
      "errorText": "",
      "validation": {
        "required": true
      }
    },
    {
      "name": "templateSet",
      "value": null,
      "col": "col-md-6 col-lg-6",
      "type": "dropdown",
      "multiple": false,
      "label": "Template Set",
      "keyText": "title",
      "display": false,
      "data": [],
      "errorText": "",
      "validation": {
        "required": false
      }
    },
    {
      "name": "generateXml",
      "value": true,
      "col": "col-md-6 col-lg-6",
      "type": "checkbox",
      "labelPlacement": "end",
      "label": "Generate XML",
      "errorText": "",
      "validation": {
        "required": false
      }
    },
    {
      "name": "isBeforetvApplicable",
      "value": false,
      "col": "col-md-6 col-lg-6",
      "type": "checkbox",
      "labelPlacement": "end",
      "label": "Apply Before TV Setting",
      "errorText": "",
      "validation": {
        "required": false
      }
    }
  ],
  "placeholderGlobalFields": [
    [
      {
        "name": "country",
        "type": "dropdownAsync",
        "col": "col-md-4 col-lg-4",
        "value": [],
        "multiple": true,
        "label": "Group / Country",
        "keyText": "title",
        "path": "user/country-group",
        "groupBy":"group",
        "data": [],
        "errorText": "",
        "validation": {
          "required": false
        }
      },
      {
        "name": "dayPart",
        "value": null,
        "col": "col-md-4 col-lg-4",
        "type": "dropdown",
        "multiple": true,
        "label": "Day Part",
        "groupBy":"group",
        "keyText": "title",
        "data": [],
        "errorText": "",
        "validation": {
          "required": false
        }
      },
      {
        "name": "globalEpisodeReleaseDate",
        "type": "date",
        "col": "col-md-4 col-lg-4",
        "value": "",
        "label": "Release Date",
        "errorText": "",
        "validation": {
          "required": true
        }
      }
    ]
  ],
  "placeholderGlobalFieldsArr": [
    {
      "name": "country",
      "type": "dropdown",
      "col": "col-md-4 col-lg-4",
      "value": [],
      "multiple": true,
      "label": "Group / Country",
      "keyText": "title",
      "path": "user/country-group",
      "groupBy":"group",
      "data": [],
      "errorText": "",
      "validation": {
        "required": false
      }
    },
    {
      "name": "dayPart",
      "value": null,
      "col": "col-md-4 col-lg-4",
      "type": "dropdown",
      "multiple": true,
      "groupBy":"group",
      "label": "Day Part",
      "keyText": "title",
      "data": [],
      "errorText": "",
      "validation": {
        "required": false
      }
    },
    {
      "name": "globalEpisodeReleaseDate",
      "type": "date",
      "col": "col-md-4 col-lg-4",
      "value": "",
      "label": "Release Date",
      "errorText": "",
      "validation": {
        "required": true
      }
    }
  ]
}
