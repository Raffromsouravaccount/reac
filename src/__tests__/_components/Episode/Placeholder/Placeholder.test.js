
import React from 'react';
import { shallow, ShallowWrapper, configure } from 'enzyme';
import moxios from 'moxios';

import { findByTestAttr } from '../../../../Utils';
import { constantText } from '../../../../_helpers/constants.text';
import axios from "../../../../_helpers/axiosInstance";

import Placeholder from '../../../../_components/Episode/Placeholder/Placeholder';
import BreadcrumbsComp from "../../../../_components/Common/BreadCrumbs/BreadCrumbs";

/**
 * Factory function to create a ShallowWrapper for the License Component.
 * @function setup
 * @param {object} props - Component props specific to this setup.
 * @param {object} state - Initial state for setup.
 * @returns {ShallowWrapper}
 */
const baseProps = {
  match: {
    params: { id: 'xyz123', seasonId: 'test123' }
  }
}
const setup = (props = {}) => {
  const wrapper = shallow(<Placeholder {...baseProps} />);
  return wrapper;
}


describe("Placeholder Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setup({...baseProps});
    moxios.install(axios)
  });
  afterEach(() => {
    moxios.uninstall(axios);
  });

  it("Should call componentDidMount()", () => {
    const instance = wrapper.instance();
    instance.componentDidMount();
    wrapper.setState({ seasonId: 134 });
  })

  it("Should call getPlaceholderData API", (done) => {
    const response = {
      response: {
        status: 201,
        data: {
          lastDate: null,
          totalEpisode: "1"
        }
      }
    }
    wrapper.instance().getPlaceholderData();
    moxios.wait(function () {
      const request = moxios.requests.mostRecent();
      request.respondWith(response).then((res) => {
        expect(wrapper.state().totalEpisode).toBe("1");
        done()
      });
    })
  })

  it("Should Render Placeholder Component", () => {
    expect(wrapper.exists()).toBe(true);
    // expect(wrapper).toMatchSnapshot();
  })

  it("Should Render Placeholder Wrapper", async () => {
    const placeholderWrapper = findByTestAttr(wrapper, 'placeholderWrapper');
    expect(placeholderWrapper.exists()).toBe(true);
  })

  it("Should Render Placeholder BreadcrumbsComp", async () => {
    const component = wrapper.find(BreadcrumbsComp);
    expect(component.length).toBe(1);
  })

  it("Should Render Placeholder Heading", async () => {
    const headingWrapper = findByTestAttr(wrapper, 'headingWrapper');
    const placeholderHeading = findByTestAttr(wrapper, 'placeholderHeading');
    expect(headingWrapper.exists()).toBe(true);
    expect(placeholderHeading.text()).toMatch(constantText.placeholderConstant.placeholder);
  })

  it("Should Render Placeholder Episode Property Heading", async () => {
    const episodePropertyWrapper = findByTestAttr(wrapper, 'episodePropertyWrapper');
    const episodePropertiesHeading = findByTestAttr(wrapper, 'episodePropertiesHeading');
    expect(episodePropertyWrapper.exists()).toBe(true);
    expect(episodePropertiesHeading.text()).toMatch(constantText.placeholderConstant.episodeProperties);
  })

  it("Should Render Placeholder Episode Property Form", async () => {
    const episodePropertiesForm = findByTestAttr(wrapper, 'episodePropertiesForm');
    expect(episodePropertiesForm.exists()).toBe(true);
  })

  it("Should Render Placeholder Global Property Form", async () => {
    const globalPropertyWrapper = findByTestAttr(wrapper, 'globalPropertyWrapper');
    const globalPropertyHeading = findByTestAttr(wrapper, 'globalPropertyHeading');
    const globalPropertiesForm = findByTestAttr(wrapper, 'globalPropertiesForm');
    expect(globalPropertyWrapper.exists()).toBe(true);
    expect(globalPropertyHeading.text()).toMatch(constantText.placeholderConstant.globalFieldsPlaceholder);
    expect(globalPropertiesForm.exists()).toBe(true);
  })

  it('should check formatData method', async () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'formatData');
    let shallowArr = [
      {
        "name": "assetType",
        "value": null,
        "col": "col-md-6 col-lg-6",
        "type": "dropdownAsync",
        "multiple": false,
        "label": "Asset Type",
        "keyText": "title",
        "path": "/master/EpisodeSubType",
        "data": [],
        "errorText": "",
        "validation": {
          "required": true
        }
      },
      {
        "name": "episodeNoStartFrom",
        "value": "",
        "col": "col-md-6 col-lg-6",
        "type": "text",
        "label": "Episode No start from",
        "helperText": "",
        "errorText": "",
        "validation": {
          "required": true,
          "charCountLimit": 250,
          "minValue": "",
          "maxLength": 300,
          "isDigit": true,
          "isAlphaNumericWithSpecialChars": true
        }
      },
    ]
    console.log(shallowArr)
    instance.formatData('episodeProperties', shallowArr);
    expect(instance.formatData).toHaveBeenCalled();
  })

})