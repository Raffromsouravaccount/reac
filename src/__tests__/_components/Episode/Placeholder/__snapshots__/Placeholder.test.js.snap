// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Component Should Render Placeholder Component 1`] = `
<div
  className="d-wrap c-n"
  data-test="placeholderWrapper"
>
  <BreadcrumbsComp
    links={
      Array [
        Object {
          "color": "inherit",
          "route": "/dashboard",
          "text": "Dashboard",
        },
        Object {
          "color": "inherit",
          "route": "/tvshow",
          "text": "TV Shows",
        },
        Object {
          "color": "inherit",
          "route": "/tvshow/edit/undefined",
          "text": "Update TV Show",
        },
        Object {
          "color": "inherit",
          "route": "/tvshow/edit/undefined/season",
          "text": "Seasons",
        },
        Object {
          "color": "inherit",
          "route": "/tvshow/edit/undefined/season/undefined",
          "text": "Update Season",
        },
      ]
    }
    typography={
      Array [
        Object {
          "color": "textPrimary",
          "text": "Placeholder",
        },
      ]
    }
  />
  <div
    className="profile-head flex align-items-center justify-content-between"
    data-test="headingWrapper"
  >
    <div
      className="back-user-btn"
    >
      <span
        onClick={[Function]}
      >
        <test-file-stub />
      </span>
      <strong>
        <span
          data-test="placeholderHeading"
        >
          Placeholder
        </span>
      </strong>
    </div>
  </div>
  <div
    className="col-12"
    data-test="episodePropertyWrapper"
  >
    <div
      className="row"
    >
      <div
        className="whitebox col-md-9 col-lg-9 col-xl-10"
      >
        <div
          className="row"
        >
          <div
            className="ccm-head w-100 flex align-items-center justify-content-between m-b-30"
          >
            <h4
              data-test="episodePropertiesHeading"
            >
              Episode Properties
            </h4>
          </div>
        </div>
        <div
          className="episode-properties"
        >
          <div
            className="row input-space-35"
          >
            <FormRender
              data-test="episodePropertiesForm"
              form={
                Array [
                  Object {
                    "col": "col-md-6 col-lg-6",
                    "data": Array [],
                    "errorText": "",
                    "keyText": "title",
                    "label": "Asset Type",
                    "multiple": false,
                    "name": "assetType",
                    "path": "/master/EpisodeSubType",
                    "touched": 0,
                    "type": "dropdownAsync",
                    "valid": true,
                    "validation": Object {
                      "required": true,
                    },
                    "value": null,
                  },
                  Object {
                    "col": "col-md-6 col-lg-6",
                    "errorText": "",
                    "helperText": "Your Next episode no. should be  11",
                    "label": "Episode No start from",
                    "name": "episodeNoStartFrom",
                    "touched": 0,
                    "type": "text",
                    "valid": true,
                    "validation": Object {
                      "charCountLimit": 250,
                      "isAlphaNumericWithSpecialChars": true,
                      "isDigit": true,
                      "maxLength": 300,
                      "minValue": 2,
                      "required": true,
                    },
                    "value": "",
                  },
                  Object {
                    "col": "col-md-6 col-lg-6",
                    "data": Array [
                      Object {
                        "group": "weekdays",
                        "id": "monday",
                        "title": "Monday",
                      },
                      Object {
                        "group": "weekdays",
                        "id": "tuesday",
                        "title": "Tuesday",
                      },
                      Object {
                        "group": "weekdays",
                        "id": "wednesday",
                        "title": "Wednesday",
                      },
                      Object {
                        "group": "weekdays",
                        "id": "thursday",
                        "title": "Thursday",
                      },
                      Object {
                        "group": "weekdays",
                        "id": "friday",
                        "title": "Friday",
                      },
                      Object {
                        "group": "weekend",
                        "id": "saturday",
                        "title": "Saturday",
                      },
                      Object {
                        "group": "weekend",
                        "id": "sunday",
                        "title": "Sunday",
                      },
                    ],
                    "errorText": "",
                    "groupBy": "group",
                    "helperText": "If not selected, release date will be same for all Assets",
                    "keyText": "title",
                    "label": "Day Part",
                    "multiple": true,
                    "name": "dayPart",
                    "touched": 0,
                    "type": "dropdown",
                    "valid": true,
                    "validation": Object {
                      "required": true,
                    },
                    "value": Array [],
                  },
                  Object {
                    "col": "col-md-6 col-lg-6",
                    "errorText": "",
                    "label": "Number of Assets",
                    "name": "numberOfAssets",
                    "touched": 0,
                    "type": "text",
                    "valid": true,
                    "validation": Object {
                      "charCountLimit": 250,
                      "isAlphaNumericWithSpecialChars": true,
                      "isDigit": true,
                      "maxLength": 300,
                      "required": true,
                    },
                    "value": "",
                  },
                  Object {
                    "col": "col-md-6 col-lg-6",
                    "errorText": "",
                    "helperText": "Your next release date should be 23 Mar 2021",
                    "label": "Release Date",
                    "minDateValue": "2021-03-23",
                    "name": "releaseDate",
                    "touched": 0,
                    "type": "date",
                    "valid": true,
                    "validation": Object {
                      "required": true,
                    },
                    "value": "",
                  },
                  Object {
                    "col": "col-md-6 col-lg-6",
                    "errorText": "",
                    "label": "Generate XML",
                    "labelPlacement": "end",
                    "name": "generateXml",
                    "touched": 0,
                    "type": "checkbox",
                    "valid": true,
                    "validation": Object {
                      "required": false,
                    },
                    "value": false,
                  },
                  Object {
                    "col": "col-md-6 col-lg-6",
                    "errorText": "",
                    "label": "Do you want to create duplicate of the asset",
                    "labelPlacement": "end",
                    "name": "duplicate",
                    "touched": 0,
                    "type": "checkbox",
                    "valid": true,
                    "validation": Object {
                      "required": false,
                    },
                    "value": false,
                  },
                ]
              }
              handleAutoCreateInput={[Function]}
              handleBlur={[Function]}
              isDisable={false}
              onChange={[Function]}
              selectGroup={[Function]}
              setSelectDataArr={[Function]}
            />
          </div>
          <div
            className="global-wrap"
            data-test="globalPropertyWrapper"
          >
            <div
              className="placeholder-title"
            >
              Global Fields
            </div>
            <div
              className="global-row"
              key="0"
            >
              <div
                className="add-plush-row top-title"
              >
                <div
                  className="top-text"
                  data-test="globalPropertyHeading"
                >
                  Global Field Placeholder Set 
                  1
                </div>
                <div
                  className="add-another-f-btn create-btn"
                >
                  <div
                    className="auto-button-add add-btn-field flex align-items-center"
                    id="addRemoveMultipleFieldsBtn"
                    onClick={[Function]}
                  >
                    <span
                      className="plush-icon-btn"
                    />
                  </div>
                </div>
              </div>
              <div
                className="row input-space-35"
              >
                <FormRender
                  data-test="globalPropertiesForm"
                  form={
                    Array [
                      Object {
                        "col": "col-md-4 col-lg-4",
                        "data": Array [],
                        "errorText": "",
                        "groupBy": "group",
                        "keyText": "title",
                        "label": "Group / Country",
                        "multiple": true,
                        "name": "country",
                        "path": "user/country-group",
                        "type": "dropdownAsync",
                        "validation": Object {
                          "required": false,
                        },
                        "value": Array [],
                      },
                      Object {
                        "col": "col-md-4 col-lg-4",
                        "data": Array [
                          Object {
                            "group": "weekdays",
                            "id": "monday",
                            "title": "Monday",
                          },
                          Object {
                            "group": "weekdays",
                            "id": "tuesday",
                            "title": "Tuesday",
                          },
                          Object {
                            "group": "weekdays",
                            "id": "wednesday",
                            "title": "Wednesday",
                          },
                          Object {
                            "group": "weekdays",
                            "id": "thursday",
                            "title": "Thursday",
                          },
                          Object {
                            "group": "weekdays",
                            "id": "friday",
                            "title": "Friday",
                          },
                          Object {
                            "group": "weekend",
                            "id": "saturday",
                            "title": "Saturday",
                          },
                          Object {
                            "group": "weekend",
                            "id": "sunday",
                            "title": "Sunday",
                          },
                        ],
                        "errorText": "",
                        "groupBy": "group",
                        "keyText": "title",
                        "label": "Day Part",
                        "multiple": true,
                        "name": "dayPart",
                        "type": "dropdown",
                        "validation": Object {
                          "required": false,
                        },
                        "value": null,
                      },
                      Object {
                        "col": "col-md-4 col-lg-4",
                        "errorText": "",
                        "label": "Release Date",
                        "name": "globalEpisodeReleaseDate",
                        "type": "date",
                        "validation": Object {
                          "required": true,
                        },
                        "value": "",
                      },
                    ]
                  }
                  groupIndex={0}
                  handleAutoCreateInput={[Function]}
                  handleBlur={[Function]}
                  isDisable={false}
                  onChange={[Function]}
                  selectGroup={[Function]}
                  setSelectDataArr={[Function]}
                />
              </div>
            </div>
          </div>
          <ButtonField
            buttonText="Save"
            className="zee-btn-field zee-half"
            color="primary"
            disabled={true}
            onClick={[Function]}
            variant="contained"
          />
        </div>
      </div>
      <div
        className="col-md-3 col-lg-3 col-xl-2"
      >
        <QuickLinks
          className=""
          customText="You can Multiple Assets from this Section"
          header_text="Quick Links"
          showTips={true}
        />
      </div>
    </div>
  </div>
  <CommonModel
    btn1Action={[Function]}
    btn1Text="Yes"
    btn2Action={[Function]}
    className="popup-wrap status-popup"
    des=""
    showBtn1={true}
    showBtn2={true}
    showDes={true}
    showIcon={false}
    showTitle={true}
    state={false}
  />
</div>
`;
